### PR TITLE
Fixed and added some sidebar buttons

### PR DESCRIPTION
**Pull request by Perseus333**
- Made some xpaths to some sidebar buttons more robust by relying on hrefs instead of index numbers.
- Fixed the shorts button so that it works on firefox aswell
- Added a toggle to disable the courses button in the sidebar

### CODE REVIEW

File: scripts/content.js [hunk 1]
```diff
@@ -100,49 +100,56 @@ const DEFAULT_ELEMENTS = [
   },
   {
     id: "my-channel",
-    selector: "(//div[@id='section-items']/ytd-guide-entry-renderer)[1]",
+    selector: "//ytd-guide-entry-renderer[a[starts-with(@href, '/@')]]",
```
Comment (Pedro-Gregorio) [inline scripts/content.js:103]:
```suggestion
    selector: "//div[@id='section-items']/ytd-guide-entry-renderer[a[starts-with(@href, '/@')]]",
```

**Review by Pedro-Gregorio — CHANGES_REQUESTED** [review]
Otherwise we're also hiding the Subscriptions channels, and that isn't ideal.

**Comment by Perseus333** [comment]
Thanks for spotting that, I guess that during testing I also had the Subscriptions channels hidden so I didn't notice.

**Comment by Perseus333** [comment]
Now it has support for hiding the "Your movies & TV" section in the sidebar thanks to https://github.com/Pedro-Gregorio/TubeMod/issues/40, and https://github.com/Pedro-Gregorio/TubeMod/issues/40#issuecomment-2380576180 for providing the href. This is currently untested but it *should* work.

**Comment by Rashnain** [comment]
The "Hide Courses button" has also the wrong id, it acutally has `for="my-videos"` instead of `for="courses"`

**Comment by Rashnain** [comment]
When fixed the "Hide Your movies & TV Button" works well. I can't test for "Hide Courses Button" because I don't have it.

**Comment by Perseus333** [comment]
Should be fixed, it also happened to my-movies-and-tv. Thansk for pointing it out

**Comment by Perseus333** [comment]
> When fixed the "Hide Your movies & TV Button" works well. I can't test for "Hide Courses Button" because I don't have it.

It works for me.

**Comment by Pedro-Gregorio** [comment]
@Perseus333 thank you for this! Adding it to V1.5.0 :)
I don't have the movies and tv button, but I'll trust @Rashnain review on that end!

@Perseus333 would you mind sharing a print screen of the sidebar (without anything hidden :D) just to ensure the order is correct.

**Comment by Perseus333** [comment]
> @Perseus333 would you mind sharing a print screen of the sidebar (without anything hidden :D) just to ensure the order is correct.

Here you go (btw, I don't know how to make this images smaller, sorry for the size):
image

**Comment by Pedro-Gregorio** [comment]
Wait, I'm confused. Where's the "Your movies & TV" button? 🤔

**Comment by Perseus333** [comment]
> Wait, I'm confused. Where's the "Your movies & TV" button? 🤔

I don't know. But it should either be above or below courses. I don't think there is anyone that has both. Although I can provide a link to a course so that someone can add to their library so that they have access to the "courses" sidebar button and report which is the correct order.

**Comment by Pedro-Gregorio** [comment]
My bad, just found the issue where you and Rashnain talked about it :)
Once again, thank you both!

**Review by Pedro-Gregorio — APPROVED** [review]

**Pull request merged**